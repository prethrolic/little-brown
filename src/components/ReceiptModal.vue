<template>
  <div v-if='isOpen' class='rcm-backdrop'>
    <h2 class='rcm-title'>Thank You for Shopping with Us!</h2>
    <h3 class='rcm-subtitle'>here's your receipt</h3>
    <div class='rcm-receipt-container'>
      <div class='rcm-receipt-body'>
        <h3 class='rcm-store-title'>Little Brown</h3>
        <div class='rcm-information'>
          <div v-html='date.toString()' />
          <div>Service Cashier: {{ user.displayName }}</div>
        </div>
        <div class='rcm-divider' />
        <div class='rcm-product-info' v-for='id in Object.keys(cart)' :key='id'>
          <div v-html='cart[id].book.title' />
          <div v-html='cart[id].count' />
          <div v-html='cart[id].book.price * cart[id].count' style='text-align: right'/>
        </div>
        <div class='rcm-text'><br>Total {{ cartCount }} items</div>
        <div class='rcm-divider' />
        <div class='rcm-payment-info'>
          <div>Subtotal</div>
          <div v-html='paymentInfo.subtotal' />
        </div>
        <div class='rcm-payment-info'>
          <div>Discount</div>
          <div v-html="'- ' + paymentInfo.discount" />
        </div>
        <div class='rcm-payment-info'>
          <div>Sales Tax</div>
          <div>+ 0.00</div>
        </div>
        <div class='rcm-divider' />
        <div class='rcm-payment-info large'>
          <div>Total</div>
          <div v-html="'à¸¿ '+ paymentInfo.total" />
        </div>
        <div class='rcm-divider' />
        <div class='rcm-payment-info'>
          <div>Cash</div>
          <div v-html='paymentInfo.cash' />
        </div>
        <div class='rcm-payment-info'>
          <div>Change</div>
          <div v-html='(paymentInfo.cash - paymentInfo.total).toFixed(2)' />
        </div>
      </div>
    </div>
    <div class='rcm-button-container'>
      <button class='rcm-button cancel' @click="$emit('bill:cancel')">
        Cancel Bill
      </button>
      <button class='rcm-button confirm' @click="$emit('bill:success')">
        Proceed
      </button>
    </div>
  </div>
</template>
<script>
export default {
  name: 'ReceiptModal',
  props: {
    date: Date,
    user: Object,
    cart: Object,
    paymentInfo: Object,
    isOpen: Boolean,
  },
  computed: {
    cartCount() {
      let count = 0;
      Object.keys(this.cart).forEach((id) => {
        count += this.cart[id].count;
      });

      return count;
    },
  },
};
</script>
<style lang="scss" scoped>
@import '@/assets/scss/variables.scss';

.rcm {
  &-backdrop {
    position: fixed;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    background-color: white;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }

  &-receipt-container {
    width: 30%;
    height: 60%;
    max-width: 500px;
    min-width: 400px;
    max-height: 600px;
    background-color: $primary;
    padding: 32px 24px;
    margin: 24px 0 8px 0;
    overflow-x: auto;
    overflow-y: scroll;
    border-radius: 8px;
    display: flex;
    align-items: center;
  }

  &-receipt-body {
    width: 80%;
    margin: auto;
    padding: 24px 16px 36px 16px;
    border-radius: 12px;
    background: white;
  }

  &-store-title {
    margin: 8px 0;
  }

  &-information {
    font-size: 12px;
    color: $dark-gray;
  }

  &-divider {
    width: 100%;
    margin: 16px auto;
    border-top: 1px dashed $dark-gray;
  }

  &-product-info {
    display: grid;
    grid-gap: 16px;
    grid-template-columns: 1fr .1fr .2fr;
    text-align: left;

    & div {
      display: -webkit-box;
      -webkit-line-clamp: 1;
      -webkit-box-orient: vertical;
      overflow-y: hidden;
      font-size: 10px;
    }
  }

  &-text {
    font-size: 10px;
    text-align: left;
  }

  &-payment-info {
    display: flex;
    justify-content: space-between;
    font-size: 10px;

    &.large {
      font-size: 14px;
    }
  }

  &-button-container {
    width: 30%;
    min-width: 400px;
    margin-left: auto;
    margin-right: auto;
    display: flex;
    justify-content: space-around;
  }

  &-button {
    display: flex;
    align-items: center;
    justify-content: center;
    margin: 24px 0;
    border: none;
    height: 64px;
    width: 40%;
    min-width: fit-content;
    padding: 0 24px;
    border-radius: 12px;
    font-size: 20px;
    font-weight: bold;
    transition: 0.2s ease-in-out;
    cursor: pointer;

    &.confirm{
      background-color: $secondary;
      color: white;

      &:hover {
        background-color: $secondary-hover;
      }
    }

    &.cancel {
      background-color: $light-gray;
      color: $dark-gray;

      &:hover {
        background-color: $light-gray-hover;
      }
    }
  }
}

@media(max-width: 768px) {
  .rcm {
    &-receipt-container {
      width: 80%;
      min-width: 0;
    }

    &-information {
      font-size: 8px;
    }

    &-button-container {
      width: 50%;
      min-width: 300px;
    }

    &-button {
      margin: 24px 0;
      border: none;
      height: 48px;
      width: 150px;
      font-size: 14px;
    }
  }
}

@media(max-width: 600px) {
  .rcm {
    &-receipt-container {
      width: 80%;
      min-width: 0;
    }

    &-button-container {
      width: 80%;
    }
  }
}
</style>
