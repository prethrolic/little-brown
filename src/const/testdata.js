export const bookList = [
  {
    "cover": "https://d1w7fb2mkkr3kw.cloudfront.net/assets/images/book/mid/9781/4088/9781408855652.jpg",
    "price": "350",
    "title": "Harry Potter and the Philosopher's Stone (I)",
    "id": "9781408855652"
  },
  {
    "cover": "https://d1w7fb2mkkr3kw.cloudfront.net/assets/images/book/mid/9781/4088/9781408855669.jpg",
    "price": "350",
    "title": "Harry Potter and the Chamber of Secrets (II)",
    "id": "9781408855669"
  },
  {
    "cover": "https://d1w7fb2mkkr3kw.cloudfront.net/assets/images/book/mid/9781/4088/9781408855676.jpg",
    "price": "340",
    "title": "Harry Potter and the Prisoner of Azkaban (III)",
    "id": "9781408855676"
  },
  {
    "cover": "https://d1w7fb2mkkr3kw.cloudfront.net/assets/images/book/mid/9780/2419/9780241975367.jpg",
    "price": "230",
    "title": "The Monk of Mokha",
    "id": "9780241975367"
  },
  {
    "cover": "https://d1w7fb2mkkr3kw.cloudfront.net/assets/images/book/mid/9781/7847/9781784706883.jpg",
    "price": "300",
    "title": "The Language of Kindness",
    "id": "9781784706883"
  },
];

export const cart = {
  "9781408855652": {
    book: {
    "cover": "https://d1w7fb2mkkr3kw.cloudfront.net/assets/images/book/mid/9781/4088/9781408855652.jpg",
    "price": "350",
    "title": "Harry Potter and the Philosopher's Stone (I)",
    "id": "9781408855652"
    },
    count: 2,
  },
  "9781408855669": {
    book: {
      "cover": "https://d1w7fb2mkkr3kw.cloudfront.net/assets/images/book/mid/9781/4088/9781408855669.jpg",
      "price": "350",
      "title": "Harry Potter and the Chamber of Secrets (II)",
      "id": "9781408855669"
    },
    count: 1,
  },
  "9781408855676": {
    book: {
      "cover": "https://d1w7fb2mkkr3kw.cloudfront.net/assets/images/book/mid/9781/4088/9781408855676.jpg",
      "price": "340",
      "title": "Harry Potter and the Prisoner of Azkaban (III)",
      "id": "9781408855676"
    },
    count: 1,
  },
  "9780241975367": {
    book: {
      "cover": "https://d1w7fb2mkkr3kw.cloudfront.net/assets/images/book/mid/9780/2419/9780241975367.jpg",
      "price": "230",
      "title": "The Monk of Mokha",
      "id": "9780241975367"
    },
    count: 2,
  },
};

export const inPromotion = ["9781408855652", "9781408855669", "9781408855676"];

export const user = {
  displayName: 'John Doe',
  pictureUrl: 'https://pbs.twimg.com/profile_images/1269585418107805697/6QKUkoWJ_400x400.jpg',
};

export const paymentInfo = {
  subtotal: '1850.00',
  discount: '114.40',
  total: '1735.60',
  cash: '2000',
};