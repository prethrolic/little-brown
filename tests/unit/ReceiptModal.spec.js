import { shallowMount } from '@vue/test-utils';
import ReceiptModal from '@/components/ReceiptModal.vue';
import { cart, user, paymentInfo } from '@/const/testdata.js';

const date = new Date(1466424490000);

describe('ReceiptModal.vue', () => {
  it('not render when isOpen is false', () => {
    const wrapper = shallowMount(ReceiptModal, {
      propsData: {
        isOpen: false,
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('render correctly with cart and payment information', () => {
    const wrapper = shallowMount(ReceiptModal, {
      propsData: {
        date,
        cart,
        user,
        paymentInfo,
        isOpen: true,
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});