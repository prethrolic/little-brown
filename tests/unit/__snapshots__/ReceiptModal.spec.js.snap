// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReceiptModal.vue not render when isOpen is false 1`] = `<!---->`;

exports[`ReceiptModal.vue render correctly with cart and payment information 1`] = `
<div
  class="rcm-backdrop"
>
  <h2
    class="rcm-title"
  >
    Thank You for Shopping with Us!
  </h2>
   
  <h3
    class="rcm-subtitle"
  >
    here's your receipt
  </h3>
   
  <div
    class="rcm-receipt-container"
  >
    <div
      class="rcm-receipt-body"
    >
      <h3
        class="rcm-store-title"
      >
        Little Brown
      </h3>
       
      <div
        class="rcm-information"
      >
        <div>
          Mon Jun 20 2016 19:08:10 GMT+0700 (Indochina Time)
        </div>
         
        <div>
          Service Cashier: John Doe
        </div>
      </div>
       
      <div
        class="rcm-divider"
      />
       
      <div
        class="rcm-product-info"
      >
        <div>
          Harry Potter and the Philosopher's Stone (I)
        </div>
         
        <div>
          2
        </div>
         
        <div
          style="text-align: right;"
        >
          700
        </div>
      </div>
      <div
        class="rcm-product-info"
      >
        <div>
          Harry Potter and the Chamber of Secrets (II)
        </div>
         
        <div>
          1
        </div>
         
        <div
          style="text-align: right;"
        >
          350
        </div>
      </div>
      <div
        class="rcm-product-info"
      >
        <div>
          Harry Potter and the Prisoner of Azkaban (III)
        </div>
         
        <div>
          1
        </div>
         
        <div
          style="text-align: right;"
        >
          340
        </div>
      </div>
      <div
        class="rcm-product-info"
      >
        <div>
          The Monk of Mokha
        </div>
         
        <div>
          2
        </div>
         
        <div
          style="text-align: right;"
        >
          460
        </div>
      </div>
       
      <div
        class="rcm-text"
      >
        <br />
        Total 6 items
      </div>
       
      <div
        class="rcm-divider"
      />
       
      <div
        class="rcm-payment-info"
      >
        <div>
          Subtotal
        </div>
         
        <div>
          1850.00
        </div>
      </div>
       
      <div
        class="rcm-payment-info"
      >
        <div>
          Discount
        </div>
         
        <div>
          - 114.40
        </div>
      </div>
       
      <div
        class="rcm-payment-info"
      >
        <div>
          Sales Tax
        </div>
         
        <div>
          + 0.00
        </div>
      </div>
       
      <div
        class="rcm-divider"
      />
       
      <div
        class="rcm-payment-info large"
      >
        <div>
          Total
        </div>
         
        <div>
          à¸¿ 1735.60
        </div>
      </div>
       
      <div
        class="rcm-divider"
      />
       
      <div
        class="rcm-payment-info"
      >
        <div>
          Cash
        </div>
         
        <div>
          2000
        </div>
      </div>
       
      <div
        class="rcm-payment-info"
      >
        <div>
          Change
        </div>
         
        <div>
          264.40
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="rcm-button-container"
  >
    <button
      class="rcm-button cancel"
    >
      
      Cancel Bill
    
    </button>
     
    <button
      class="rcm-button confirm"
    >
      
      Proceed
    
    </button>
  </div>
</div>
`;
